File: libs/models/src/common/paged-result-model.ts
import { z } from 'zod';

export type PagedResultModel<T> = {
  items: T[];
  pageNumber: number;
  pageSize: number;
  totalCount: number;
};

export function pagedResultModel<T extends z.ZodTypeAny>(schema: T) {
  return z.object({
    pageSize: z.number(),
    pageNumber: z.number(),
    items: z.array(schema),
    totalCount: z.number(),
  });
}

File: apps/blazor-web/Shared/Models/PagedResultModel.cs
namespace UpliftLunchNLearn.BlazorWeb.BlazorWeb.Shared.Models;

public record PagedResultModel<T>(T[] Items, int PageSize, int PageNumber, int TotalCount)
{
  public static PagedResultModel<T> Empty()
  {
    return new PagedResultModel<T>(Array.Empty<T>(), 10, 1, 0);
  }
}

public record PagedQuery(int PageSize, int PageNumber);
